### PR TITLE
first commit

### PR DESCRIPTION
**Pull request by jeffbum**
Initial commit of this tool.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,2 +1,56 @@
...
+    - [Resources](#resources)
+    - [Download](#download)
+    - [How to use the cli](#how-to-use-the-cli)
+    - [Example Usage](#example-usage)
+  - [Linting](#linting)
+  - [Changelog](#changelog)
+
+## Overview
+
+The purpose of the reference application is to provide a starting point for Node.js development that encapsulates good tools and best practices such as TypeScript, configuration management, linting, testing, and documentation.
```
Comment (korymarks) [inline README.md:17]:
```suggestion
The purpose of the reference application is to provide a starting point for Node.js development that encapsulates good tools and best practices such as JavaScript, configuration management, linting, testing, and documentation.
```

File: index.js [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+    return DBMLDefiniton;
+  });
+  convertToDBML.unshift(`Table ${tableName} {`);
+  convertToDBML.push('}\n\n');
+  const returnValue = convertToDBML.join('\n');
+  o && console.log(`creating/adding to: ${schemaName}.dbml to your output path with the dbml definition of table ${tableName}.`)
+  o ? fs.appendFile(`${o || './'}${schemaName}.dbml`, returnValue, 'utf8', () => {}) : console.log(returnValue);
+}
+
+function tableColumnInfoQueryFunc(tableNameArr, schemaName) {
```
Comment (korymarks) [inline index.js]: function names should be verbs... maybe getTableColumn()?

File: index.js [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+  tableNameArr.forEach(tableName => {
+    const query = tableColumnInfoQuery(tableName);
+    dbClient.query(query, (err, res) => {
+      if (err) throw new Error(err);
+      writeToDBML(res.rows, tableName, schemaName);
+    });
+  });
+}
+
+function tablesInSchemaFunc(schemaNameArr) {
```
Comment (korymarks) [inline index.js:79]: function name to be an action. What do you think about `getTablesInSchema`?

File: index.js [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+    const schemaNameQuery = tablesInSchemaQuery(schemaName);
+    dbClient.query(schemaNameQuery, (err, res) => {
+      if (err) throw new Error(err);
+      const tableNameArr = res.rows.map(row => row.tablename);
+      tableColumnInfoQueryFunc(tableNameArr, schemaName);
+    });
+  });
+}
+
+function schemaQueryFunc() {
```
Comment (korymarks) [inline index.js:95]: an action name for this function

**Review by korymarks — CHANGES_REQUESTED** [review]
- [ ] should refactor the database connection string to not have database name or remove the --db arg for the database name. If you do not supply the database name on the connection string and supply it only in the --db then it says it ran but it doesn't... no logs or loaded just to the last message. 
- [ ] It might be nice to be able to supply schema as an option and only get that returned in a .dbml file
- [ ] I didn't see any error handling as mentioned above and from our conversation yesterday
- [ ] If the file already exists on the repo, you should not just append. This seems like a bug to fix.
- [ ] add *.dbml to the .gitignore would be a good practice

**Comment by jeffbum** [comment]
Closing this PR, as I opened another on the feature branch to master. Will be deleting develop branch as well.

**Pull request closed without merging**